script.js

```diff
@@ -57,21 +57,197 @@ form.addEventListener("submit", (e) => {
   e.preventDefault();
   console.log("Form submitted");
 
-  const selectedEmotion = document.querySelector('input[name="feeling"]:checked').value;
+  const selectedRadio = document.querySelector('input[name="feeling"]:checked');
+  if (selectedRadio) {
+    const selectedEmotion = selectedRadio.value;
+    emotions[selectedEmotion]++;
+    emotions["total"]++;
 
-  emotions[selectedEmotion]++;
-  emotions["total"]++;
+    // Saves the number to the localStorage
+    localStorage.setItem("emotions", JSON.stringify(emotions));
 
-  //Saves the number to the localStorage
-  localStorage.setItem("emotions", JSON.stringify(emotions));
+    // Update the percentages in the HTML
+    const total = emotions["total"];
+    updatePercentages(emotions, total);
 
-  // Update the percentages in the HTML
-  const total = emotions["total"];
-  updatePercentages(emotions, total);
-
-  // Reset the form to be unchecked
-  document.querySelector('input[name="feeling"]:checked').checked = false;
+    // Reset the radio button to be unchecked
+    selectedRadio.checked = false;
+  } else {
+    console.log("No option selected");
+  }
 });
 
+
 // Update percentages when page loads
-document.addEventListener('DOMContentLoaded', updatePercentages(emotions, emotions["total"]), false);
\ No newline at end of file
+document.addEventListener('DOMContentLoaded', updatePercentages(emotions, emotions["total"]), false);
+
+
+// Function to load emotions from local storage (also to initialize emotions)
+function loadEmotions() {
+  const storedEmotion = localStorage.getItem("emotions");
+  if (storedEmotion !== null) {
+      return JSON.parse(storedEmotion);
+  } else {
+      return {
+          upset: 0,
+          unhappy: 0,
+          neutral: 0,
+          happy: 0,
+          joyful: 0,
+          total: 0
+      }
+  }
+}
+
+//TESTING - I have commented out the method of actually calling the test since we dont have to do testing every time the app is run
```

Comment:
Tests look good! Could you move them to a separate JS file of just the tests that could be run separately (this is how testing is typically done)